### PR TITLE
Add MAPL v2.22.0 and extdata2g variant

### PR DESCRIPTION
**Pull request by kgerheiser**
Corresponding to https://github.com/NOAA-EMC/hpc-stack/pull/473

### CODE REVIEW

File: var/spack/repos/jcsda-emc/packages/mapl/package.py [hunk 1]
```diff
@@ -47,6 +48,7 @@ class Mapl(CMakePackage):
     variant('esma_gfe_namespace', default=True)
     variant('shared', default=True)
     variant('debug', default=False, description='Make a debuggable version of the library')
+    variant('extdata2g', default=True, description='Use ExtData2G')
```
Comment (climbfuji) [inline var/spack/repos/jcsda-emc/packages/mapl/package.py]: I missed the conversation on hpc-stack, what does this thing do?

**Comment by climbfuji** [comment]
And now there is a merge conflict (because of the last merge from the release branch). Should be easy to fix.

**Comment by climbfuji** [comment]
> And now there is a merge conflict (because of the last merge from the release branch). Should be easy to fix.

Fixed.

File: var/spack/repos/jcsda-emc/packages/mapl/package.py (hunk 1 above)
Comment (kgerheiser) [reply var/spack/repos/jcsda-emc/packages/mapl/package.py]: Whether or not to use the new ExtData (2G - 2nd Generation) component in MAPL for GOCART. This option was disabled by default, but now it's enabled.

Comment (kgerheiser) [reply var/spack/repos/jcsda-emc/packages/mapl/package.py]: I just set it to be disabled by default. I was going to do it in packages.yaml, but for our purposes it's better to do here.

**Review by climbfuji — APPROVED** [review]
I guess the new default (False) is what you want.

**Pull request merged**